dask_expr/groupby.py

```diff
@@ -285,9 +275,27 @@ def __init__(
         sort=None,
         observed=None,
         dropna=None,
+        slice=None,
     ):
+        by_ = by if isinstance(by, (tuple, list)) else [by]
+        self._slice = slice
+        # Check if we can project columns
+        projection = None
+        if (
+            np.isscalar(slice)
+            or isinstance(slice, (str, list, tuple))
+            or (
+                (is_index_like(slice) or is_series_like(slice))
+                and not is_dask_collection(slice)
+            )
+        ):
+            projection = set(by_).union(
+                {slice} if (np.isscalar(slice) or isinstance(slice, str)) else slice
+            )
+            projection = [c for c in obj.columns if c in projection]
+
         self.by = [by] if np.isscalar(by) else list(by)
-        self.obj = obj
+        self.obj = obj[projection] if projection is not None else obj
```

Comment:
Perhaps the implied column projection could also be added in `_simplify_down` (i.e. add a `Projection` if `self.obj` includes more columns than necessary). It seems like it could be nice to separate out the optimization from the core logic. Does that make sense?

Reply:
I think so, will check this tomorrow

Reply:
Rick and I chatted offline. We can add this step to ``SingleAggregation`` and ``GroupbyAggregation`` but not to ``GroupBy``, since it's not an Expression. not a blocker here and will tackle in a follow up